script/staging/StagingDeploymentBase.sol

```diff
@@ -130,117 +138,139 @@ contract StagingDeploymentBase is Script {
-        susds = new MockSUsds(address(usds));
+        ScriptTools.exportContract(mainnet.name, "dai",     dai);
+        ScriptTools.exportContract(mainnet.name, "sUsds",   susds);
+        ScriptTools.exportContract(mainnet.name, "usdc",    USDC);
+        ScriptTools.exportContract(mainnet.name, "usds",    usds);
+        ScriptTools.exportContract(mainnet.name, "daiUsds", daiUsds);
+        ScriptTools.exportContract(mainnet.name, "psm",     psm);
+    }
 
-        // Step 3: Deploy mocked MCD contracts
+    function _setUpMocks() internal {
+        require(IERC20(USDC).balanceOf(deployer) >= USDC_UNIT_SIZE * 10, "USDC balance too low");
 
-        vat      = new MockVat(mainnet.admin);
-        usdsJoin = new MockUsdsJoin(mainnet.admin, address(vat), address(usds));
-        daiUsds  = new MockDaiUsds(mainnet.admin, address(dai), address(usds));
-        jug      = new MockJug();
-        psm      = new MockPSM(mainnet.admin, address(usdc), address(dai));
+        dai   = address(new MockERC20("DAI",  "DAI",  18));
+        usds  = address(new MockERC20("USDS", "USDS", 18));
+        susds = address(new MockSUsds(usds));
 
-        // Step 4: Seed relevant contracts with tokens
+        daiUsds = address(new MockDaiUsds(mainnet.admin, dai, usds));
+        psm     = address(new MockPSM(mainnet.admin, USDC, dai));
 
         // Mint USDS into the join contract
-        usds.mint(address(usdsJoin), USDS_UNIT_SIZE);
+        MockERC20(usds).mint(usdsJoin, USDS_UNIT_SIZE);
 
         // Fill the psm with dai and usdc
-        usdc.transfer(address(psm), USDC_UNIT_SIZE * 10);
-        dai.mint(address(psm), USDS_UNIT_SIZE);
+        IERC20(USDC).transfer(psm, USDC_UNIT_SIZE * 10);
+        MockERC20(dai).mint(psm, USDS_UNIT_SIZE);
 
         // Fill the DaiUsds contract with both tokens
-        dai.mint(address(daiUsds), USDS_UNIT_SIZE);
-        usds.mint(address(daiUsds), USDS_UNIT_SIZE);
+        MockERC20(dai).mint(daiUsds, USDS_UNIT_SIZE);
+        MockERC20(usds).mint(daiUsds, USDS_UNIT_SIZE);
+    }
 
-        vm.stopBroadcast();
+    function _useDeployedContracts() internal {
+        dai     = mainnet.config.readAddress(".dai");
+        usds    = mainnet.config.readAddress(".usds");
+        susds   = mainnet.config.readAddress(".susds");
+        daiUsds = mainnet.config.readAddress(".daiUsds");
+        psm     = mainnet.config.readAddress(".psm");
 
-        // Step 5: Export all deployed addresses
+        require(IERC20(usds).balanceOf(deployer) >= USDS_UNIT_SIZE, "USDS balance too low");
 
-        ScriptTools.exportContract(mainnet.name, "dai",   address(dai));
-        ScriptTools.exportContract(mainnet.name, "sUsds", address(susds));
-        ScriptTools.exportContract(mainnet.name, "usdc",  address(usdc));
-        ScriptTools.exportContract(mainnet.name, "usds",  address(usds));
+        // Transfer USDS into the join contract
+        IERC20(usds).transfer(usdsJoin, USDS_UNIT_SIZE);
```

Comment:
`usdsJoin` is not set

Reply:
Moved to mint usds into the deployer in `_setUpMocks` and then transfer USDS in the `_setUpMCDMocks` section